packaging/arch/PKGBUILD

```diff
@@ -184,4 +184,15 @@ package_snapd-git() {
   # and make sure that target exists so that we don't have dangling symlinks
   # after installing the package
   install -d -m 755 "$pkgdir/var/lib/snapd/snap"
+
+  # pre-create directories
+  install -d -m 755 "$pkgdir/var/cache/snapd"
+  install -d -m 755 "$pkgdir/var/lib/snapd/assertions"
+  install -d -m 755 "$pkgdir/var/lib/snapd/desktop/applications"
+  install -d -m 755 "$pkgdir/var/lib/snapd/device"
+  install -d -m 755 "$pkgdir/var/lib/snapd/hostfs"
+  install -d -m 755 "$pkgdir/var/lib/snapd/mount"
+  install -d -m 755 "$pkgdir/var/lib/snapd/seccomp/bpf"
+  install -d -m 755 "$pkgdir/var/lib/snapd/snap/bin"
+  install -d -m 755 "$pkgdir/var/lib/snapd/snaps"
```

Comment:
This list is not complete, e.g. `/var/lib/snapd/cache`, `/var/lib/snapd/cookie`, `/var/lib/snapd/firstboot` are missing - I'm not sure how important it is for the list to be complete here. Note that some of these dirs have more strict permissions.

Reply:
Yeah, +1 on the patch but add more directories please.

Reply:
Some confusion here, need advice. The list in here: https://github.com/snapcore/snapd/blob/master/packaging/ubuntu-16.04/snapd.dirs is slightly different from https://github.com/snapcore/snapd/blob/master/packaging/opensuse-42.2/snapd.spec#L268 and https://github.com/snapcore/snapd/blob/master/packaging/fedora/snapd.spec#L602.

I've captured this in a list:
```
snap                                # classic snaps
var/cache/snapd                     # all 
usr/lib/snapd                       # all
var/lib/snapd/apparmor/snap-confine # apparmor only
var/lib/snapd/auto-import           # only ubuntu?
var/lib/snapd/desktop               # all?
var/lib/snapd/environment           # only ubuntu?
var/lib/snapd/firstboot             # only ubuntu?
var/lib/snapd/lib/gl                # all?
var/lib/snapd/snaps/partial         # all?
var/lib/snapd/void                  # all, perms 000
var/snap                            # global data, all? 

# fedora
%dir %{_sharedstatedir}/snapd
%dir %{_sharedstatedir}/snapd/assertions
%dir %{_sharedstatedir}/snapd/desktop
%dir %{_sharedstatedir}/snapd/desktop/applications
%dir %{_sharedstatedir}/snapd/device
%dir %{_sharedstatedir}/snapd/hostfs
%dir %{_sharedstatedir}/snapd/mount
%dir %{_sharedstatedir}/snapd/seccomp
%dir %{_sharedstatedir}/snapd/seccomp/bpf
%dir %{_sharedstatedir}/snapd/snaps
%dir %{_sharedstatedir}/snapd/snap
%ghost %dir %{_sharedstatedir}/snapd/snap/bin
%dir %{_localstatedir}/cache/snapd
%dir %{_localstatedir}/snap
%attr(0000,root,root) %{_sharedstatedir}/snapd/void
# missing (all?)
# -  /var/lib/snapd/lib/gl?


# openSUSE
%dir %attr(0000,root,root) /var/lib/snapd/void
%dir /snap
%dir /snap/bin
%dir %{_libexecdir}/snapd
%dir /var/lib/snapd
%dir /var/lib/snapd/apparmor
%dir /var/lib/snapd/apparmor/profiles
%dir /var/lib/snapd/apparmor/snap-confine
%dir /var/lib/snapd/assertions
%dir /var/lib/snapd/desktop
%dir /var/lib/snapd/desktop/applications
%dir /var/lib/snapd/device
%dir /var/lib/snapd/hostfs
%dir /var/lib/snapd/mount
%dir /var/lib/snapd/seccomp
%dir /var/lib/snapd/seccomp/bpf
%dir /var/lib/snapd/snaps
%dir /var/cache/snapd
# missing (all?)
# - /var/snap?
# - /var/lib/snapd/lib/gl?
```